source/slang/slang-ir-glsl-legalize.cpp

```diff
@@ -3729,4 +3729,97 @@ void legalizeDispatchMeshPayloadForGLSL(IRModule* module)
     });
 }
 
+void legalizeDynamicResourcesForGLSL(CodeGenContext* context, IRModule* module)
+{
+    List<IRGlobalParam*> toRemove;
+    Dictionary<IRType*, IRGlobalParam*> aliasedParams;
```

Comment:
The dictionary is keyed on a single IRType, this seems incorrect if there are more than one DynamicResource global parameter. Two distinct dynamic resource global params should map to distinct aliased global params, so this dictionary needs to be cleared after processing each original global param, or be changed to key on a IRGlobalParam+IRType pair.

Reply:
Fixed